crates/matrix-sdk-crypto/src/types/one_time_keys.rs

```diff
@@ -98,29 +98,60 @@ impl SignedKey {
 }
 
 /// A one-time public key for "pre-key" messages.
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
 #[serde(untagged)]
 pub enum OneTimeKey {
     /// A signed Curve25519 one-time key.
     SignedKey(SignedKey),
 
     /// An unsigned Curve25519 one-time key.
-    #[serde(deserialize_with = "deserialize_curve_key", serialize_with = "serialize_curve_key")]
+    #[serde(serialize_with = "serialize_curve_key")]
     Key(Curve25519PublicKey),
+}
 
-    /// An unknown one-time key type.
-    Unknown(Value),
+impl OneTimeKey {
+    /// Deserialize the [`OneTimeKey`] from a [`DeviceKeyAlgorithm`] and a Raw
+    /// JSON value.
+    ///
+    /// The [`DeviceKeyAlgorithm`] is needed because multiple algorithms might
+    /// have the same JSON representation.
```

Comment:
Should this be a regular comment on the parameter instead?

It seems more like an implementation note than something that users should know about.

Reply:
I'll just remove it.